src/core_modules/capture-core/src/reducers/descriptions/currentSelections.reducerDescriptionGetter.js

```diff
@@ -1,20 +1,46 @@
 // @flow
+import programs from 'capture-core/metaDataMemoryStores/programCollection/programCollection';
 import { createReducerDescription } from '../../trackerRedux/trackerReducer';
 import type { Updaters } from '../../trackerRedux/trackerReducer';
 import { actionTypes as mainSelectionsActionTypes } from '../../components/Pages/MainPage/mainSelections.actions';
+import { actionTypes as setCurrentSelectionsActionTypes } from '../../components/QuickSelector/actions/QuickSelector.actions';
 import {
     actionTypes as newEventSelectionActionTypes,
 } from '../../components/Pages/NewEvent/newEventSelections.actions';
 import {
     actionTypes as editEventActionTypes,
 } from '../../components/Pages/EditEvent/editEvent.actions';
+import {
+    actionTypes as mainPageSelectorActionTypes,
+} from '../../components/Pages/MainPage/MainPageSelector/MainPageSelector.actions';
+import {
+    actionTypes as editEventPageSelectorActionTypes,
+} from '../../components/Pages/EditEvent/EditEventSelector/EditEventSelector.actions';
+import {
+    actionTypes as newEventPageSelectorActionTypes,
+} from '../../components/Pages/NewEvent/NewEventSelector/NewEventSelector.actions';
 
 type CurrentSelectionsState = {
     programId?: ?string,
     orgUnitId?: ?string,
+    categories?: ?Object,
 };
 
 const calculateCompleteStatus = (state: CurrentSelectionsState) => {
+    const programsArray = Array.from(programs.values());
+    const selectedProgram = programsArray.find(program => program.id === state.programId);
+
+    if (state.programId && selectedProgram.categoryCombo && !selectedProgram.categoryCombo.isDefault) {
```

Comment:
we should remove complexity by following the pattern for metadata already established. I'll show you.

Reply:
Done